src/System.Net.Security/src/System/Net/Unix/SecuritySafeHandles.cs

```diff
@@ -19,24 +19,31 @@ internal sealed class SafeFreeCertContext : DebugSafeHandle
     internal sealed class SafeFreeCertContext : SafeHandle
     {
 #endif
-        public SafeFreeCertContext(IntPtr handle) : base(handle, true)
+        private readonly SafeX509Handle _certificate;
+
+        public SafeFreeCertContext(SafeX509Handle certificate) : base(IntPtr.Zero, true)
         {
+            if ((null != certificate) && !certificate.IsInvalid)
+            {
+                _certificate = Interop.libcrypto.X509_dup(certificate);
```

Comment:
My opinion is that S.N.Security PAL should not rely on how Interop.OpenSsl implements its methods. Unless you feel strongly about this, I would like to leave this as "won't fix"